### DISCUSSION TITLE
spring_mount for joint

### DISCUSSION THREAD

**Discussion by wooseokH** [raisimTech/raisimLib#245] [Q&A]
Hello!

I am conducting a simulation(python) using manipulator. I want to use "spring_mount" function. I wonder how the direction(Vec<4>) is defined in spring_mount. When the spring is connected as shown in the figure(black line) below, how should the direction be defined?

![spring_SEA](https://user-images.githubusercontent.com/96863807/148329956-d8acb020-8ce0-4996-a53b-7efc8c314d67.png)

**Comment by jhwangbo** [comment]
If you have a ball joint, it is a reference quaternion. For any other joint types, you only have to use the first element, which represents angle or position, depending on the joint type

**Reply by wooseokH** [reply]
thank you for answer

but, i am still confused about the direction setting of spring_mount. I don't know what happens when "x 0 0 0 " x is 0, and also x is changes. 
and, when spring_mount is used, can i still use the joint torque command "setGeneralizedForce"?

Lastly, When learning SEA-based robots(Anymal) by simulation, have you considered SEA(spring)? Can spring_mount be that role?

**Reply by jhwangbo** [reply]
The spring mount means the position at where one end of the spring is mounted. The other one is mounted on the moving link. If you have a joint angle of 5 and the spring mount is 2, then the spring force is k(5-2).

the spring force is ``k(x-x_mount)``. The direction is trivial. if x > x_mount, then the force is positive and vice versa.
Yes, you can use the torque command.
You can simulate SEA using a RaiSim spring.

**Reply by wooseokH** [reply]
Similar results were obtained when simulated based on what you told me.

However, the spring mount is given only as a constant value as urdf. I want to make sure that the spring is deformed only when the load force is applied, but I couldn't. Is there any other way?

And,there are APIs such as addSpring, getSpring().. , but it can not work in my raisim(raisimpy) should I update the risim library? or can't i use it on raisimpy(python)?

**Reply by jhwangbo** [reply]
I added a method to change the spring mount in python. check raisimPy/examples/springs.py

**Reply by wooseokH** [reply]
It helped me a lot. Thank you.
